src/onyx/datomic/client.clj

```diff
@@ -0,0 +1,101 @@
+(ns onyx.datomic.client
+  (:require [clojure.string :as str]
+            [datomic.client.api :as d]
+            [onyx.datomic.protocols :as dp]
+            [taoensso.timbre :as log])
+  (:import [java.net URI]))
+
+(def tempid-counters (atom -1000000))
+
+(defn- _cloud-client [{:keys [datomic-cloud/system
+                              datomic-cloud/region
+                              datomic-cloud/query-group
+                              datomic-cloud/endpoint
+                              datomic-cloud/proxy-port] :as datomic-config}]
+  (let [region (or region (System/getenv "DATOMIC_CLOUD_REGION"))
+        system (or system (System/getenv "DATOMIC_CLOUD_SYSTEM"))
+        proxy-port (or proxy-port (System/getenv "DATOMIC_CLOUD_PROXY_PORT"))
+        proxy-port (when (string? proxy-port) (Integer/parseInt proxy-port))]
+    (d/client  {:server-type :cloud
+                :region region
+                :system system
+                :query-group (or query-group (System/getenv "DATOMIC_CLOUD_QUERY_GROUP") system)
+                :endpoint (or endpoint (format "http://entry.%s.%s.datomic.net:8182/" system region))
+                :proxy-port (or proxy-port 8182)})))
+
+(defn- _peer-server-client [{:keys [datomic-client/access-key
+                                    datomic-client/secret
+                                    datomic-client/endpoint] :as datomic-config}]
+  (let [access-key (or access-key (System/getenv "DATOMIC_CLIENT_ACCESS_KEY"))
+        secret (or secret (System/getenv "DATOMIC_CLIENT_SECRET"))
+        endpoint (or endpoint (System/getenv "DATOMIC_CLIENT_ENDPOINT"))]
+    (d/client  {:server-type :peer-server
+                :access-key access-key
+                :secret secret
+                :endpoint endpoint})))
+
+(def cloud-client (memoize _cloud-client))
+(def peer-server-client (memoize _peer-server-client))
+
+(defn- safe-connect-cloud [{:keys [datomic-cloud/db-name] :as datomic-config}]
+  (when (nil? db-name)
+    (throw (ex-info "either :datomic-cloud/db-name is required to connect." datomic-config)))
```

Comment:
Likely shouldn't say `either` here.

Reply:
I originally included both `:datomic-client/db-name` and `:datomic-cloud/db-name`. As mentioned in README, I cannot run tests with client API for On-Prem, so I decided to take the former out, but either was left out. It's already merged, so should I create a separate PR to fix it?